patch-package
--- a/node_modules/topojson/build/topojson.js
+++ b/node_modules/topojson/build/topojson.js
@@ -242,7 +242,7 @@ function meshArcs(topology, o, filter) {
 
     geomsByArc.forEach(arguments.length < 3
         ? function(geoms) { arcs.push(geoms[0].i); }
-        : function(geoms) { if (filter(geoms[0].g, geoms[geoms.length - 1].g)) arcs.push(geoms[0].i); });
+        : function(geoms) { if (filter.apply(null, geoms.map(function(geom) { return geom.g }))) arcs.push(geoms[0].i); });
   } else {
     for (var i = 0, n = topology.arcs.length; i < n; ++i) arcs.push(i);
   }
